### PR TITLE
Addition of tests for HandleDomainCreationReplicationTask in replicationTaskExecutor

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**
Added tests in common/domain folder for HandleDomainCreationReplicationTask in replicationTaskExecutor go file

**Why?**
to increase code coverage

**How did you test it?**
Unit Tests

**Potential risks**
No risks, as it only involves tests with mocking behaviors.

### CODE REVIEW

File: common/domain/replicationTaskExecutor_test.go [hunk 1]
```diff
@@ -248,3 +249,195 @@ func TestDomainReplicationTaskExecutor_Execute(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleDomainCreationReplicationTask(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockDomainManager := persistence.NewMockDomainManager(ctrl)
```
Comment (davidporter-id-au) [inline common/domain/replicationTaskExecutor_test.go:257]: so, the problem with putting it up here is that you're sharing it between table-rows, so it's a little tricky to get the assertions

File: common/domain/replicationTaskExecutor_test.go [hunk 2]
```diff
@@ -248,3 +249,195 @@ func TestDomainReplicationTaskExecutor_Execute(t *testing.T) {
...
+					}, nil).AnyTimes() // Successfully finds a domain but with a different name
+			},
+			task:      validTask,
+			wantError: true, // Expect an error because of the name/UUID collision
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setup()
```
Comment (davidporter-id-au) [inline common/domain/replicationTaskExecutor_test.go:434]: Why not create the domain manager and pass it in to the setup function here? That way you'll be creating a new mock every table test iteration and there's no risk of two rows setting up different expectations

```
t.Run(tt.name, func(t *testing.T) {
  ctrl := gomock.NewController(t)
  defer ctrl.Finish()
  
  mockDomainManager := persistence.NewMockDomainManager(ctrl)
  tt.setup(mockDomainManager)
```

Comment (abhishekj720) [reply common/domain/replicationTaskExecutor_test.go:434]: oh right, i think i missed it since just pushed it in the morning, thanks for the catch.

Comment (abhishekj720) [reply common/domain/replicationTaskExecutor_test.go:434]: but what about validTask? 
then i will be defining it in each case?

File: common/domain/replicationTaskExecutor_test.go [hunk 3]
```diff
@@ -248,3 +250,341 @@ func TestDomainReplicationTaskExecutor_Execute(t *testing.T) {
...
+				PreviousFailoverVersion: 0,
+			},
+			wantError: true, // Expect an error because of the name/UUID collision
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockDomainManager := persistence.NewMockDomainManager(ctrl)
+			mockLogger := log.NewNoop()
```
Comment (3vilhamster) [inline common/domain/replicationTaskExecutor_test.go:573]: nit: testlogger.New

File: common/domain/replicationTaskExecutor_test.go [hunk 4]
```diff
@@ -248,3 +250,341 @@ func TestDomainReplicationTaskExecutor_Execute(t *testing.T) {
...
+			mockTimeSource := clock.NewRealTimeSource()
```
Comment (3vilhamster) [inline common/domain/replicationTaskExecutor_test.go:574]: nit: clock.NewMockedTimeSourceAt with a fixed time.

File: common/domain/replicationTaskExecutor_test.go [hunk 5]
```diff
@@ -248,3 +250,341 @@ func TestDomainReplicationTaskExecutor_Execute(t *testing.T) {
...
+				mockDomainManager.EXPECT().
+					CreateDomain(gomock.Any(), gomock.Any()).
+					Return(nil, ErrInvalidDomainStatus).Times(1)
+
+				mockDomainManager.EXPECT().
+					GetDomain(gomock.Any(), gomock.Any()).
+					Return(nil, errors.New("unexpected error")).Times(1) // Simulate an unexpected error from GetDomain
+			},
+			task: &types.DomainTaskAttributes{
+				DomainOperation: types.DomainOperationCreate.Ptr(),
```
Comment (3vilhamster) [inline common/domain/replicationTaskExecutor_test.go:486]: I suggest refactoring out test data from here. It is very hard to read the test cases that are over 200 lines with intendation.

**Review by 3vilhamster — COMMENTED** [review]
A bit hard to read the test. :(

**Comment by abhishekj720** [comment]
Already done the changes in another PR.

**Pull request closed without merging**